packages/nimble-components/src/select/specs/filerable-select-hld.md

```diff
@@ -0,0 +1,97 @@
+# Filterable Select
+
+## Problem Statement
+
+Our clients have a need of a filterable dropdown component that does not allow arbitrary text as a value, but only values available in the dropdown. The filtering will have the following behaviors (in the initial feature set):
+
+-   The filter text will match _any_ text within the `textContent` of each `ListOption` in a case insensitive way.
+-   Each time the dropdown is opened the filter text is cleared.
+-   While the dropdown is opened all keystrokes except `<ArrowUp>`. `<ArrowDown>`, `<Home>` and `<End>` will apply to the filter.
```

Comment:
How does the user select an item using only the keyboard when filtering is enabled? In the [keyboard section of the interaction spec](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/select/specs/IxD.md#dropdown) it mentions that Enter or Space should choose the currently highlighted item. But it looks like we're currently proposing that the filter will receive those keys, not the items / list.

One idea would be to simply forward those keystrokes too. I think that could make sense for Enter but I could imagine some users being surprised if they typed Space in the filter and it selected an item.

Or am I missing something about the focus states? I'm assuming the filter still has keyboard focus after you press ArrowUp/ArrowDown but maybe that's not accurate.

Reply:
I've changed it to mention that `<Enter>` will also not apply to the filter. To be clear, when filtering is enabled, `<Enter>` will commit the value (i.e. raise the `change` event), while `<Space>` will be added as part of the filter text.

You are correct about the focus states.